x/incentives/keeper/gauge.go

```diff
@@ -108,9 +109,12 @@ func (k Keeper) CreateGauge(ctx sdk.Context, isPerpetual bool, owner sdk.AccAddr
 		}
 	}
 
-	// Ensure that the denom this gauge pays out to exists on-chain
-	if !k.bk.HasSupply(ctx, distrTo.Denom) && !strings.Contains(distrTo.Denom, "osmovaloper") {
-		return 0, fmt.Errorf("denom does not exist: %s", distrTo.Denom)
+	// check if distrTo.Denom is uosmo, if not ensure that the denom this gauge
+	// pays out to exists on-chain
+	if distrTo.Denom != appParams.BaseCoinUnit {
```

Comment:
@p0mvn want to make sure this a fair assumption to make? uosmo should ALWAYS exists on-chain right?

Reply:
It is a fair assumption. However, I do not understand why we would want to have it? Why can't we check this even for uosmo

Reply:
its because in simulator we were randomly generating denoms and often times the sender wouldnot have uosmo, or not have enough supply of uosmo, and it would constantly fail here

Reply:
Make sense, ty! I think we should then get in https://github.com/osmosis-labs/osmosis/pull/4768 and remove this line.

We should avoid hacking regular code to make convenience for tests

I followed up with @phamminh0811 to refresh #4768 and merge it. 

Please ping me if it's taking longer than Monday, and I can refresh and merge it

Reply:
https://github.com/osmosis-labs/osmosis/pull/5028 wtih the relevant change is now merged

Reply:
fixed this https://github.com/osmosis-labs/osmosis/pull/5008/commits/5edb0345ba631d66d05b73f20eaa9a8a4f947b54